seml/utils.py

```diff
@@ -2,35 +2,102 @@
 import json
 import copy
 
+
 def s_if(n):
     return '' if n == 1 else 's'
 
 
-def unflatten(dictionary: dict, sep: str = '.'):
+def unflatten(dictionary: dict, sep: str = '.', recursive: bool = False, levels=None):
```

Comment:
`recursive=False` is the same as `levels=[0]`. Do we need both?

Reply:
It's actually not the same. For example, unflattening 
```python
{
"a.b.c": 111,
"a.d": {"e.f": 222},
}
```
with `recursive=False, levels=None` will yield
```python
{
"a": {
  "b": {
    "c": 111,
  }
  "d": {"e.f": 222}
}
}
```
while `recursive=True, levels=None` will produce 
```python
{
"a": {
  "b": {
    "c": 111,
  }
  "d": {
    "e: {
      "f": 222
    }
  }
}
}
```

That is, `recursive=True` tells us that we also look into (potentially flattened) sub-dictionaries and also unflatten these.

Reply:
Ah, I misunderstood this then.